pkg/operator/controller/route-status/controller_test.go

```diff
@@ -0,0 +1,197 @@
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "shard-label", "default-ic"),
+			namespace:         newNamespace("ns", "not-shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "shard-label", "", "shard-label", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", ""),
+		},
+		{
+			name:              "admitted with expression routeSelector and no namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "shard-label", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "shard-label", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "not admitted with expression routeSelector and no namespaceSelector with route with incorrect labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "not-shard-label", "default-ic"),
+			namespace:         newNamespace("ns", ""),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "shard-label", "", ""),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", ""),
+		},
+		{
+			name:              "admitted with no routeSelector and expression namespaceSelector with route with correct labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+		},
+		{
+			name:              "not admitted with no routeSelector and expression namespaceSelector with route with incorrect labels in namespace with no labels",
+			request:           reconcileRequestRoute,
+			route:             newRouteWithLabelWithAdmittedStatuses("route", "ns", "", "default-ic"),
+			namespace:         newNamespace("ns", "not-shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedRoute:     newRouteWithLabelWithAdmittedStatuses("route", "ns", ""),
+		},
+		{
+			name:              "route is nil",
+			request:           reconcileRequestRoute,
+			route:             nil,
+			namespace:         newNamespace("ns", "not-shard-label"),
+			ingressController: newIngressControllerWithSelectors("default-ic", "", "", "", "shard-label"),
+			expectedErr:       false,
```

Comment:
This is the only case that specifies `expectedErr`, and it specifies `expectedErr: false`.  Did you mean to put `expectedErr: true` here?  Do we really need `expectedErr` at all?